protocol/jsm/v2/receiver.go

```diff
@@ -0,0 +1,166 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package jsm
+
+import (
+	"context"
+	"io"
+	"sync"
+
+	"github.com/nats-io/nats.go"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+)
+
+type msgErr struct {
+	msg binding.Message
+}
+
+type Receiver struct {
+	incoming chan msgErr
+}
+
+func NewReceiver() *Receiver {
```

Comment:
what is this used for?

Reply:
NewReceiver creates a new protocol.Receiver responsible for receiving messages, an implements of protocol.Receiver